include/irr/asset/IAsset.h

```diff
@@ -167,25 +173,40 @@ class IAsset : virtual public core::IReferenceCounted
 
         virtual core::smart_refctd_ptr<IAsset> clone(uint32_t _depth = ~0u) const = 0;
 
+		inline E_MUTABILITY getMutability() const { return m_mutability; }
+		inline bool isMutable() const { return getMutability() == EM_MUTABLE; }
+		inline bool canBeConvertedToDummy() const { return (getMutability()&EM_CPU_PERSISTENT) != EM_CPU_PERSISTENT; }
+
     protected:
-        void clone_common(IAsset* _clone) const
+        inline void clone_common(IAsset* _clone) const
         {
             _clone->m_metadata = m_metadata;
             assert(!isDummyObjectForCacheAliasing);
             _clone->isDummyObjectForCacheAliasing = false;
-            _clone->m_mutable = true;
+            _clone->m_mutability = EM_MUTABLE;
         }
+		inline bool isImmutable_debug()
+		{
+			_IRR_DEBUG_BREAK_IF(!isMutable());
+			return !isMutable();
+		}
 
 	private:
 		friend IAssetManager;
 		core::smart_refctd_ptr<IAssetMetadata> m_metadata;
 
-		void setMetadata(core::smart_refctd_ptr<IAssetMetadata>&& _metadata) { m_metadata = std::move(_metadata); }
+		void setMetadata(core::smart_refctd_ptr<IAssetMetadata>&& _metadata) 
+		{
+			//we have to talk about it (TODO)
+			//if (isImmutable_debug())
+			//	return;
```

Comment:
doesn't matter what you do here because of #451 which means that it should be the `SAssetBundle` that holds the metadata, not the `IAsset` (a future TODO along with #22 #23 which are still open because of a rare scheduling loophole)

Reply:
leave a comment pointing to the github issue.

Reply:
actually https://github.com/Devsh-Graphics-Programming/Nabla/issues/14

Reply:
leave a comment